FormalConjectures/ErdosProblems/43.lean

```diff
@@ -0,0 +1,81 @@
+/-
+Copyright 2025 The Formal Conjectures Authors.
+
+Licensed under the Apache License, Version 2.0 (the "License");
+you may not use this file except in compliance with the License.
+You may obtain a copy of the License at
+
+    https://www.apache.org/licenses/LICENSE-2.0
+
+Unless required by applicable law or agreed to in writing, software
+distributed under the License is distributed on an "AS IS" BASIS,
+WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+See the License for the specific language governing permissions and
+limitations under the License.
+-/
+
+import FormalConjectures.Util.ProblemImports
+
+/-!
+# Erdős Problem 43: Disjoint Differences in Sidon Sets
+
+Let `f(N)` denote the maximum possible size of a Sidon set in `{1, ..., N}`.
+
+Suppose `A, B ⊆ {1, ..., N}` are both Sidon sets and their difference sets are disjoint:
+\[
+(A - A) ∩ (B - B) = ∅
+\]
+
+Is it true that
+\[
+\binom{|A|}{2} + \binom{|B|}{2} ≤ \binom{f(N)}{2} + O(1) ?
+\]
+
+Furthermore, if `|A| = |B|`, can this be improved to
+\[
+\binom{|A|}{2} + \binom{|B|}{2} ≤ (1 - c) \binom{f(N)}{2}
+\]
+for some constant `c > 0`?
+
+Find the link to the problem [here](https://www.erdosproblems.com/43).
+
+-/
+
+/-- The maximum size of a Sidon set in `{1, ..., N}`. -/
+noncomputable def maxSidonSetSize (N : ℕ) : ℕ :=
+  sSup {n | ∃ A : Finset ℕ, A ⊆ Finset.Icc 1 N ∧ IsSidon A.toSet ∧ A.card = n}
```

Comment:
Perhaps one could make this computable with something like:
```
instance: DecidablePred fun A : Finset ℕ ↦ IsSidon A.toSet := by
  unfold IsSidon
  unfold DecidablePred
  intro A
  simp [Decidable]
  infer_instance

def SidonSubsetsSizes (n : ℕ) :=
  Finset.image Finset.card <| (Finset.Icc 1 n).powerset.filter (fun (A : Finset ℕ) => IsSidon A.toSet)

lemma SidonSubsetsSizesNonempty (n : ℕ) : (SidonSubsetsSizes n).Nonempty:= by
  use 0
  unfold SidonSubsetsSizes
  unfold IsSidon
  aesop

def maxSidonSetSize' (n : ℕ) := (SidonSubsetsSizes n).max' (SidonSubsetsSizesNonempty n)
```

All of this should then go into `FormalConjectures/ForMathlib/Combinatorics/Basic.lean`